typescript/cli/src/config/agent.ts

```diff
@@ -19,19 +23,80 @@ export async function createAgentConfig({
   out,
 }: {
   context: CommandContext;
-  chains: string[];
+  chains?: string[];
   out: string;
 }) {
   logBlue('\nCreating agent config...');
 
-  const { registry, multiProvider, chainMetadata } = context;
+  const { registry, multiProvider, chainMetadata, skipConfirmation } = context;
   const addresses = await registry.getAddresses();
 
-  const core = HyperlaneCore.fromAddressesMap(addresses, multiProvider);
+  await handleNoChainsProvided(skipConfirmation, chains);
 
-  const startBlocks = await promiseObjAll(
-    objMap(addresses, async (chain, _) => {
-      // If the index.from is specified in the chain metadata, use that.
+  const chainAddresses = filterChainAddresses(addresses, chains);
+
+  const core = HyperlaneCore.fromAddressesMap(chainAddresses, multiProvider);
+  const startBlocks = await getStartBlocks(chainAddresses, core, chainMetadata);
+
+  if (!skipConfirmation) {
+    await handleMissingInterchainGasPaymaster(chainAddresses);
+  }
+
+  const agentConfig: AgentConfig = buildAgentConfig(
+    chains ?? Object.keys(chainAddresses),
```

Comment:
iiuc because of `filterChainAddresses`, we may be able to just pass in `Object.keys(chainAddresses)` here?